pytorch_lightning/utilities/data.py

```diff
@@ -180,7 +180,25 @@ def get_len(dataloader: DataLoader) -> Union[int, float]:
 def _update_dataloader(dataloader: DataLoader, sampler: Sampler, mode: Optional[RunningStage] = None) -> DataLoader:
     dl_kwargs = _get_dataloader_init_kwargs(dataloader, sampler, mode=mode)
     dl_cls = type(dataloader)
-    dataloader = dl_cls(**dl_kwargs)
+    try:
+        dataloader = dl_cls(**dl_kwargs)
+    except TypeError as e:
+        # improve exception message due to an incorrect implementation of the `DataLoader` where multiple subclass
+        # `__init__` arguments map to one `DataLoader.__init__` argument
+        import re
```

Comment:
the import is specially included here?

Reply:
just a micro-optimization. import times add up on the scale of large libraries like lightning